### PR TITLE
comments implementation

### PR DESCRIPTION
**Pull request by Ben-893**
Posts can have comments

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -10,18 +10,30 @@
...
 
   # These are extensions that must be enabled in order to support this database
   enable_extension "plpgsql"
 
+
+  create_table "comments", force: :cascade do |t|
+    t.text "body"
+    t.integer "post_id"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
```
Comment (archmagos) [inline db/schema.rb]: This block needs to be closed, which is possibly causing the problem. Can you add an 'end' here and then try another push?

**Comment by archmagos** [comment]
Top work fixing the failing tests, cheers Ben. Looks like the only thing that's failing is our coverage - are you able to put in a test to cover the create method in the comments controller?

**Comment by JohnForster** [comment]
Nice work ben! Get rid of that extra line that codeclimate is freaking out about and we can pull it into the master!

**Comment by Ben-893** [comment]
@JohnForster I've also added the user can post comments functionality, could you look at that too pls. Tally Ho guv! :beer: :gb:

File: app/views/posts/index.html.erb [hunk 2]
```diff
@@ -1,13 +1,27 @@
...
+    <p><%= f.submit "Create Comment" %></p>
+  <% end %>
+</div>
 
-<%= link_to new_post_path do %>
-  New post
+<% post.comments.each do |comment| %>
+  <% if comment.body %>
+    <p id="messageBox2"><%= comment.body %></p>
+    <p id="time"><%= comment.created_at.strftime("%m-%e-%y, %H:%M") %> by <%= comment.poster_name %>
```
Comment (JohnForster) [inline app/views/posts/index.html.erb]: d/m/y in the uk m8 ;)

File: db/schema.rb [hunk 3]
```diff
@@ -35,8 +36,7 @@
     t.string "message"
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
-    t.bigint "user_id"
-    t.index ["user_id"], name: "index_posts_on_user_id"
+    t.integer "user_id"
```
Comment (JohnForster) [inline db/schema.rb]: What's the story behind this change?

File: db/schema.rb [hunk 4]
```diff
@@ -50,5 +50,4 @@
 
   add_foreign_key "likes", "posts"
   add_foreign_key "likes", "users"
-  add_foreign_key "posts", "users"
```
Comment (JohnForster) [inline db/schema.rb]: Why has this line been deleted?

File: db/schema.rb (hunk 3 above)
Comment (Ben-893) [reply db/schema.rb]: When I pulled down from master, the migration you had added was blank for some reason, so when I ran db:migrate, it didn't add userid to posts. So I added in the userid column to posts but didn't do it as a foreign key.

Comment (JohnForster) [reply db/schema.rb]: Ah I think this is the same issue as Layth had. There was a problem with one of the migration files. I have done a separate pull request just fixing this issue.

**Pull request merged**